cirq/api/google/v2/run_context.proto

```diff
@@ -0,0 +1,115 @@
+syntax = "proto3";
+
+package cirq.api.google.v2;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "RunContextProto";
+option java_multiple_files = true;
+
+// The context for running a quantum program.
+message RunContext {
+  // The parameters for operations in a program.
+  repeated ParameterSweep parameter_sweeps = 1;
+}
+
+// Specifies how to repeatedly sample a circuit, with or without sweeping over
+// varying parameter-dicts.
+message ParameterSweep {
+  // How many times to sample, for each parameter-dict that is swept over.
+  // This must be set to a value strictly greater than zero.
+  int32 repetitions = 1;
+
+  // Which parameters, that control gates in the circuit, to try.
+  //
+  // The keys of the parameters generated by this sweep must be a superset
+  // of the keys in the program's operation's Args. When this is not specified,
+  // no parameterization is assumed (and the program must have no
+  // args with symbols).
+  Sweep sweep = 2;
+}
+
+// A sweep over all of the parameters in a program.
+message Sweep {
+
+  // The sweep is either a function of sweeps or an actual assignment of
+  // a symbol to a list of values.
+  oneof sweep {
+    SweepFunction sweep_function = 1;
+    SingleSweep single_sweep = 2;
+  }
+}
+
+// A function that takes multiple sweeps and produces more sweeps.
+message SweepFunction {
```

Comment:
Seems like this could be modeled with Zip and Mul messages or an enum, as we do for `SingleSweep`. I don't imagine this will evolve as fast as the gates. But I could be persuaded if you think there are other sweep functions we would want, or want to be flexible just in case.

Reply:
Moved to enum. 

The only other one that comes to mind is concatenate.